pkg/svc/codeset.go

```diff
@@ -84,5 +85,9 @@ func (s *codesetsrvc) Get(ctx context.Context, p *codeset.GetPayload) (res *code
 
 func (s *codesetsrvc) Delete(ctx context.Context, p *codeset.DeletePayload) error {
 	s.logger.Print("codeset.delete")
+	workflows := s.workflowMgr.List(ctx, nil)
+	for _, wf := range workflows {
+		s.workflowMgr.UnassignFromCodeset(ctx, wf.Name, p.Project, p.Name)
+	}
```

Comment:
This is the only thing I don't agree with. We've kept implementation details out of the codeset REST service so far. It only acts as a dumb wrapper that converts data structures domain-to-rest or vice-versa and calls the codeset store to do all the work.

This has to go into the Delete store method, which also means adding the workflow manager as a dependency for that component.

Reply:
As discussed, this will probably be a common problem for other components and so it was agreed that the solution should be proposed in another PR.